instances.tf

```diff
@@ -140,24 +146,22 @@ resource "vsphere_virtual_machine" "icpmaster" {
     }
   }
 
+  # Specify the ssh connection
+  connection {
+    user          = "${var.ssh_user}"
+    password      = "${var.ssh_password}"
+  }
+
   provisioner "file" {
     source      = "${path.module}/scripts"
     destination = "/tmp/terraform_scripts"
-
-    connection {
-      user          = "${var.ssh_user}"
-      private_key   = "${file(var.ssh_keyfile)}"
-    }
   }
 
   provisioner "remote-exec" {
-    connection {
-      user          = "${var.ssh_user}"
-      private_key   = "${file(var.ssh_keyfile)}"
-    }
-
     inline = [
       "sudo chmod u+x /tmp/terraform_scripts/*.sh",
+      "/tmp/terraform_scripts/add-public-ssh-key.sh \"${tls_private_key.ssh.public_key_openssh}\"",
+      "/tmp/terraform_scripts/add-private-ssh-key.sh \"${tls_private_key.ssh.private_key_pem}\" \"${var.ssh_user}\"",
```

Comment:
in this step we're copying the private key to every master, when I think just the boot node or boot master really needs the private key.

in addition, for security reasons we tried not to copy private keys around like this.  it might be better to generate a key locally, then use ssh-agent from the terraform machine to pass the key across SSH connections rather than copying the private keys to the filesystem.

Reply:
@jkwong888 good call. Will look into the boot master part. As far as not copying the keys, my main question is how are we going to make this work from the CAM perspective?

Reply:
@jkwong888 Ok, but I thought that's what I was achieving in this PR. Unless there is something I am missing. Also, how can I determine which is the master boot node where so I put the private ssh key only on that node?

Reply:
boot node can be a separate node or the first master node ... you can use terraform to project just the first entry in the list to make sure just the first master gets the pub key